st2api/st2api/controllers/resource.py

```diff
@@ -220,6 +238,18 @@ def _get_by_name(self, resource_name, exclude_fields=None):
 
         return resource_db
 
+    def _get_by_pack_ref(self, pack_ref, exclude_fields=None):
+        try:
+            resource_db = self.access.get(pack=pack_ref, exclude_fields=exclude_fields)
+        except Exception:
+            resource_db = None
+
+        if not resource_db:
```

Comment:
It's just an existing code pattern I followed (yeah, I could just move throw inside the except block).